### PR TITLE
Add documenatation on how to create an API server

### PR DESCRIPTION
**Pull request by SimonLab**
ref: #132
Adding instruction on how to create an API server with Phoenix

### CODE REVIEW

File: api-server.md [hunk 1]
```diff
@@ -69,4 +69,40 @@ where
...
+we need to let the server allow these request.
+By default only the requests from the same domain name as the server are allowed.
+
+We can use the [cors_plug](https://hex.pm/packages/cors_plug) package to
+update the `:api` pipeline to allow other domain names.
+
+- Add the dependency `{:cors_plug, "~> 2.0"}` to `mix.exs` file and run `mix deps.get`
+- Add in `endpoint.ex` file `plug CORSPlug, origin: ["*"]`
 
+see https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
```
Comment (nelsonic) [inline api-server.md]: 👍

**Comment by nelsonic** [comment]
@SimonLab I'm trying to follow this guide now, but cannot find the **`code`** ... 🤷 
Did you push the end state of your example App to GitHub? (_I tried checking your [repos](https://github.com/SimonLab?tab=repositories) ..._)

**Comment by SimonLab** [comment]
I've mostly tested how to create an API only application with https://github.com/dwyl/app-api/
The router doesn't have any `:browser` pipeline and only use the `:api` one: https://github.com/dwyl/app-api/blob/master/lib/app_api_web/router.ex

There is also a CORS configuration on the `endpoint.ex` file:
https://github.com/dwyl/app-api/blob/d077e3a42b0e99273fd1650560ef102faa161ff2/lib/app_api_web/endpoint.ex#L43

**Comment by nelsonic** [comment]
@SimonLab thanks for clarifying. 👍

**Pull request merged**